Reorganize the Join properties dialog description (#4500)
qgis/QGIS-Documentation: source/docs/user_manual/working_with_vector/vector_properties.rst

Joins Properties
================

> |join| The :guilabel:`Joins` tab allows you to associate to features

Reviewer: ```suggestion
|join| The :guilabel:`Joins` tab allows you to associate features
```

Author: Would that convey the idea that the current layer serves as base and the features from the join layer are pulled and connected to it?

Reviewer: Yes.  I think this is a better way to phrase it ("associate features of the current layer to features from another loaded vector layer or table").  I think we should avoid the word "join" here, since that would not explain much.

> of the current layer (called ``Target layer``) features from another

Reviewer: ```suggestion
of the current layer (called ``Target layer``) to features from another
```

> loaded vector layer. The join criteria is an attribute shared by both

Reviewer: ```suggestion
loaded vector layer (or table). The join is based on an attribute that is shared by the
```

> layers. The layers can be geometryless or not but their join attribute

Reviewer: ```suggestion
layers. The layers can be geometryless (tables) or not but their join attribute
```

   Join an attribute table to an existing vector layer

> The steps above are enough to create a join, adding to the target layer's

Reviewer: ```suggestion
The steps above will create a join, where **ALL** the attributes of the
first matching feature in the join layer is added to the target layer's feature.
```

> features **ALL** the attributes of their first matching feature in the join

Reviewer: ```suggestion
```

> layer. QGIS provides more options to tweak the join:

Reviewer: ```suggestion
QGIS provides more options to tweak the join:
```

* |checkbox| :guilabel:`Cache join layer in virtual memory`: allows you to cache
  values in memory (without geometries) from the joined layer in order to speed
  up lookups.
* |unchecked| :guilabel:`Create attribute index on the join field`
* |unchecked| :guilabel:`Dynamic form`: helps to synchronize join fields on the
  fly, according to the :guilabel:`Target field`. This way, constraints for
  join fields are also correctly updated. Note that it's deactivated by default
  because it may be very time consuming if you have a lot of features or a
  myriad of joins.
* If the target layer is editable, then some icons will be displayed in the

>   attribute table next to fields, in order to inform their status:

Reviewer: ```suggestion
  attribute table next to fields, in order to inform about their status:
```

  * |joinNotEditable|: the join layer is not configured to be
    editable. If you want to be able to edit join features from the target
    attribute table, then you have to check the option
    |checkbox| :guilabel:`Editable join layer`.
  * |joinedLayerNotEditable|: the join layer is well configured to be
    editable, but its current status is read only.

>   * |joinHasNotUpsertOnEdit|: the join layer is editable but synchronization

Reviewer: ```suggestion
  * |joinHasNotUpsertOnEdit|: the join layer is editable, but synchronization
```
